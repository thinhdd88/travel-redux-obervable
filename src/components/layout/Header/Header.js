import React, {PureComponent} from 'react';
import { Link } from 'react-router-dom';
import logo from 'assets/images/logo.png';

export class Header extends PureComponent {
    render() {
        return (
            <header className="header">
                <nav className="navbar navbar-default">
                    <div className="container">
                        <Link activeClassName="active" onlyActiveOnIndex={true} className="nav-link logo" to="/">
                            <img alt="" src={logo} />
                        </Link>
                        <Link className="nav-link" to="/test">
                            test
                        </Link>
                    </div>
                </nav>
            </header>
        );
    }
}