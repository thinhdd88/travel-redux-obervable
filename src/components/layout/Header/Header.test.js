import React from 'react';
import {shallow} from 'enzyme';
import {Header} from './Header';

describe('Header', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
    });
});
