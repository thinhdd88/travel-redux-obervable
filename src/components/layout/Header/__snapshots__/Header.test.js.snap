// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-default"
>
        <div
                className="container"
        >
                <Link
                        className="nav-link logo"
                        replace={false}
                        to="/"
                >
                        <img
                                alt=""
                                src="logo.png"
                        />
                </Link>
                <Link
                        className="nav-link"
                        replace={false}
                        to="/test"
                >
                        test
                </Link>
        </div>
</nav>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <Link
                    className="nav-link logo"
                    replace={false}
                    to="/"
          >
                    <img
                              alt=""
                              src="logo.png"
                    />
          </Link>
          <Link
                    className="nav-link"
                    replace={false}
                    to="/test"
          >
                    test
          </Link>
</div>,
        "className": "navbar navbar-default",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="nav-link logo"
              replace={false}
              to="/"
>
              <img
                            alt=""
                            src="logo.png"
              />
</Link>,
            <Link
              className="nav-link"
              replace={false}
              to="/test"
>
              test
</Link>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt=""
                src="logo.png"
/>,
              "className": "nav-link logo",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test",
              "className": "nav-link",
              "replace": false,
              "to": "/test",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-default"
>
          <div
                    className="container"
          >
                    <Link
                              className="nav-link logo"
                              replace={false}
                              to="/"
                    >
                              <img
                                        alt=""
                                        src="logo.png"
                              />
                    </Link>
                    <Link
                              className="nav-link"
                              replace={false}
                              to="/test"
                    >
                              test
                    </Link>
          </div>
</nav>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <Link
                        className="nav-link logo"
                        replace={false}
                        to="/"
            >
                        <img
                                    alt=""
                                    src="logo.png"
                        />
            </Link>
            <Link
                        className="nav-link"
                        replace={false}
                        to="/test"
            >
                        test
            </Link>
</div>,
          "className": "navbar navbar-default",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="nav-link logo"
                replace={false}
                to="/"
>
                <img
                                alt=""
                                src="logo.png"
                />
</Link>,
              <Link
                className="nav-link"
                replace={false}
                to="/test"
>
                test
</Link>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt=""
                  src="logo.png"
/>,
                "className": "nav-link logo",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test",
                "className": "nav-link",
                "replace": false,
                "to": "/test",
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
